### ISSUE TITLE
Storage: LVM root fs rename

### ISSUE THREAD

**Issue by pbuchan-rh** [cockpit-project/cockpit#20815]
### Explain what happens

1. Rename root FS LVM in cockpit storage
2. Reboot machine
3. Machine hangs unable to find root FS

I would assume hitting the edit and rename button would make the necessary system config changes to prevent the system from hanging at reboot.  If not maybe this feature needs removed or isolated to not include the root FS.

![Screenshot from 2024-07-29 10-54-32](https://github.com/user-attachments/assets/702297de-b884-4437-bc40-dd15fa4dd4a3)

### Version of Cockpit

311.2

### Where is the problem in Cockpit?

Storage

### Server operating system

Red Hat Enterprise Linux

### Server operating system version

9.4

### What browsers are you using?

Firefox, Chrome

**Comment by jelly** [comment]
@mvollmer mind taking a look?

**Comment by mvollmer** [comment]
> I would assume hitting the edit and rename button would make the necessary system config changes to prevent the system from hanging at reboot.

Yes, that would be nice. Boot loader config changes are tricky, unfortunately, but we do them in other places and we should be able to do them here as well.

> If not maybe this feature needs removed or isolated to not include the root FS.

Let's disable group renaming when the rootfs is on it.

(My current understanding is this: the initrd is able to find the root filesystem just fine without any help from a configuration option, even if it is on a LVM volume. Normally, the initrd just activates all LVM2 volume groups that it finds. However, it also has the feature to only activate a specified set of volume groups. If this feature is used to only activate the volume group with the rootfs, then we need to change the configuration when the volume group is renamed.)
